<?xml version="1.0"?>
<launch>
  <!-- start Webots -->
  <arg name="no-gui" default="false," doc="Start Webots with minimal GUI"/>
  <include file="$(find webots_ros)/launch/webots.launch">
    <arg name="mode" value="realtime"/>
    <arg name="no-gui" value="$(arg no-gui)"/>
    <arg name="world" value="$(env HOME)/projects/webots/borunte/worlds/with_robot_and_ezgripper_cupboard_barcodes.wbt"/>
  </include>

  <arg name="camera_name" default="rgb_camera" />
  <!-- frame_id for the camera -->
  <arg name="frame_id" default="rgb_camera_link" />
  <!-- By default, calibrations are stored to file://${ROS_HOME}/camera_info/${NAME}.yaml
   To use your own fill this arg with the corresponding url, e.g.:
   "file:///$(find your_camera_package)/config/your_camera.yaml" -->
  <arg name="camera_info_url" default="file:///$(find webots_arm)/config/webots_cam.yaml" />

  <node name="camera_control" pkg="webots_arm" type="camera_control" output="screen" required="true">
    <param name="frame_id" type="string" value="$(arg frame_id)" />
    <param name="camera_name" type="string" value="$(arg camera_name)" />
    <param name="image_topic" type="string" value="$(arg camera_name)/image_raw" />
    <param name="camera_info_url" type="string" value="$(arg camera_info_url)" />
    <param name="fps" type="int" value="5" />
  </node>

  <arg name="joint_update_rate" default="25" />
  <node name="$(anon joint_state_throttle)" pkg="topic_tools" type="throttle"
        args="messages joint_states $(arg joint_update_rate) joint_states_sim" required="true">
    <param name="unreliable" type="bool" value="true" />
  </node>

  <node name="joint_control_1" pkg="webots_arm" type="joint_control" output="screen" required="true">
    <param name="name" type="string" value="joint_1"/>
    <remap from="joint_states" to="joint_states_throttle"/>
  </node>
  <node name="joint_control_2" pkg="webots_arm" type="joint_control" output="screen" required="true">
    <param name="name" type="string" value="joint_2"/>
    <remap from="joint_states" to="joint_states_throttle"/>
  </node>
  <node name="joint_control_3" pkg="webots_arm" type="joint_control" output="screen" required="true">
    <param name="name" type="string" value="joint_3"/>
    <remap from="joint_states" to="joint_states_throttle"/>
  </node>
  <node name="joint_control_4" pkg="webots_arm" type="joint_control" output="screen" required="true">
    <param name="name" type="string" value="joint_4"/>
    <remap from="joint_states" to="joint_states_throttle"/>
  </node>
  <node name="joint_control_5" pkg="webots_arm" type="joint_control" output="screen" required="true">
    <param name="name" type="string" value="joint_5"/>
    <remap from="joint_states" to="joint_states_throttle"/>
  </node>
  <node name="joint_control_6" pkg="webots_arm" type="joint_control" output="screen" required="true">
    <param name="name" type="string" value="joint_6"/>
    <remap from="joint_states" to="joint_states_throttle"/>
  </node>

  <!--<node name="gripper_control" pkg="webots_arm" type="gripper_control" output="screen" required="true" /> -->

</launch>
